timesketch/api/v1/resources/graph.py

```diff
@@ -61,8 +137,38 @@ def post(self, sketch_id):
             abort(HTTP_STATUS_CODE_NOT_FOUND, 'No sketch found with this ID.')
 
         form = request.json
-        graph_name = form.get('graph_name')
-        graph_class = manager.GraphManager.get_graph(graph_name)
+        plugin_name = form.get('plugin')
+        graph_config = form.get('config')
+        refresh = form.get('refresh')
+
+        sketch_indices = [
+            timeline.searchindex.index_name
+            for timeline in sketch.active_timelines
+        ]
+
+        cache = GraphCache.get_or_create(
```

Comment:
ok, so do we want people with read access to the sketch to be able to do this? or should we add a check to see if you've go write access before proceeding?

Reply:
Yeah, for the cache it is ok for read only users to generate if it doesn't exist.